delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/directives/AuthDirectivesSpec.scala

```diff
@@ -46,34 +97,64 @@ class AuthDirectivesSpec
       }
     }
 
+  private val authorizationRoute: Route =
+    handleExceptions(IdentityError.exceptionHandler) {
+      path("user") {
+        directives.extractCaller { implicit caller =>
+          directives.authorizeFor(AclAddress.Root, permission).apply {
+            get {
+              caller match {
+                case Anonymous             => complete("anonymous")
+                case Caller(user: User, _) => complete(user.subject)
+                case _                     => complete("another")
+              }
+            }
+          }
+        }
+      }
+    }
+
   "A route" should {
 
     "correctly extract the user Alice" in {
-      Get("/user") ~> addCredentials(OAuth2BearerToken("alice")) ~> route ~> check {
+      Get("/user") ~> addCredentials(OAuth2BearerToken("alice")) ~> callerRoute ~> check {
         response.status shouldEqual StatusCodes.OK
         asString(response) shouldEqual "alice"
       }
     }
 
     "correctly extract Anonymous" in {
-      Get("/user") ~> route ~> check {
+      Get("/user") ~> callerRoute ~> check {
         response.status shouldEqual StatusCodes.OK
         asString(response) shouldEqual "anonymous"
       }
     }
 
     "fail with an invalid token" in {
-      Get("/user") ~> addCredentials(OAuth2BearerToken("bob")) ~> route ~> check {
+      Get("/user") ~> addCredentials(OAuth2BearerToken("bob")) ~> callerRoute ~> check {
         response.status shouldEqual StatusCodes.Unauthorized
       }
     }
 
     "fail with invalid credentials" in {
-      Get("/user") ~> addCredentials(BasicHttpCredentials("alice")) ~> route ~> check {
+      Get("/user") ~> addCredentials(BasicHttpCredentials("alice")) ~> callerRoute ~> check {
         response.status shouldEqual StatusCodes.Unauthorized
       }
     }
 
+    "correctly authorize user" in {
+      Get("/user") ~> addCredentials(OAuth2BearerToken("alice")) ~> authorizationRoute ~> check {
+        response.status shouldEqual StatusCodes.OK
+        asString(response) shouldEqual "alice"
+      }
+    }
+
+    "correctly reject Anonymous " in {
+      Get("/user") ~> authorizationRoute ~> check {
```

Comment:
We also should try with another user without the permission

Reply:
Added the test.